lib/ansible/modules/remote_management/cisco/cisco_ucs_dns.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/python
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cisco_ucs_dns
+short_description: Configures DNS on a Cisco UCS server.
+version_added: "2.4"
+description:
+   -  Configures DNS on a Cisco UCS server.
+options:
+    state:
+        description:
+         - if C(present), adds DNS server
+         - if C(absent), removes DNS server
+        choices: ['present', 'absent']
+        default: "present"
+    name:
+        description: ip address of dns server
+        required: true
+    descr:
+        description: description of server
+
+requirements:
+    - 'ucsmsdk'
+    - 'ucsm_apis'
+    - 'python2 >= 2.7.9 or python3 >= 3.2'
+    - 'openssl version >= 1.0.1'
+
+extends_documentation_fragment:
+    - cisco_ucs
+
+author: "Cisco Systems Inc(ucs-python@cisco.com)"
+'''
+
+
+EXAMPLES = '''
+- name: "Add DNS Server"
+  cisco_ucs_dns:
+    name: "10.10.10.10"
+    descr: "description"
+    state: "present"
+    ucs_ip: "192.168.1.1"
+    ucs_username: "admin"
+    ucs_password: "password"
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def _argument_mo():
+    return dict(
+        name=dict(required=True, type='str'),
+        descr=dict(type='str'),
+    )
+
+
+def _argument_custom():
+    return dict(
+        state=dict(default="present",
+                   choices=['present', 'absent'],
+                   type='str'),
+    )
+
+
+def _argument_connection():
+    return dict(
+        # UcsHandle
+        ucs_server=dict(type='dict'),
+
+        # Ucs server credentials
+        ucs_ip=dict(type='str'),
+        ucs_username=dict(default="admin", type='str'),
+        ucs_password=dict(type='str', no_log=True),
+        ucs_port=dict(default=None),
+        ucs_secure=dict(default=None),
+        ucs_proxy=dict(default=None)
```

Comment:
I would advice against using these ucs-prefixed names Most (if not all)l existing modules use **hostname**, **username** and **password** as the parameters for authentication. Also **ucs_port**, **ucs_secure** and **ucs_proxy** have standard parameters, the documentation names these **port**, **secure** and **proxy**.

Reply:
Prefix `ucs_` is introduced as future cisco ucs modules have parameters as hostname, username, password and port which uses the utils module cisco_ucs. So to avoid conflict we have appended prefix `ucs_`

Reply:
Well, this would be opposite of how other modules are doing it in Ansible. The default **hostname** **username** and **password** are the parameters for the module. The question is, are you doing your users a favor by turning around the logic ?

Reply:
@dagwieers: I understand its bit off from the standard names used by other ansible modules. However our modules is mapped to the inherent UCS schema so in longer run it will be useful for users too as the future modules will have parameters with names `hostname, username, password`. So as to avoid the clash in the future we opted for this `ucs_` convention.